bears/general/KeywordBear.py

```diff
@@ -2,7 +2,68 @@
 
 from coalib.bearlib import deprecate_settings
 from coalib.bears.LocalBear import LocalBear
+from coalib.results.Diff import Diff
 from coalib.results.Result import RESULT_SEVERITY, Result
+from coalib.results.SourceRange import SourceRange
+
+from bears.general.AnnotationBear import AnnotationBear
+
+
+def _get_comments(dependency_results):
+    if not dependency_results:
+        return
+
+    annotation_bear_results = dependency_results.get('AnnotationBear')
+    if (not annotation_bear_results or
+            not isinstance(annotation_bear_results, list)):
+        return
+
+    for result in annotation_bear_results:
+        yield from result.contents.get('comments', [])
+
+
+def generate_diff(comments, file, filename,
+                  line, line_number, pos):
+    diffs = {}
+    todo_source_range = SourceRange.from_values(filename, line_number,
+                                                pos + 1)
+    for comment_sourcerange in comments:
+        if todo_source_range not in comment_sourcerange:
+            continue
+
+        comment_start = comment_sourcerange.start.column
+        comment_end = comment_sourcerange.end.column
+        in_multi_line_comment = (
+            comment_sourcerange.start.line != comment_sourcerange.end.line
+        )
+
+        line_before_todo_comment = line[:comment_start - 1].rstrip()
+        line_behind_todo_comment = line[comment_end:].rstrip()
+
+        line_replacement = line_before_todo_comment + line_behind_todo_comment
+
+        diff = Diff(file)
+        if line_replacement and not in_multi_line_comment:
+
+            diff.change_line(
+                line_number,
+                line,
+                line_replacement + '\n')
+
+        elif (in_multi_line_comment and
+              line_before_todo_comment):
+            # remove just keyword, not comment
+            text_replacement = line[pos:]
+
+            diff.change_line(
+                line_number,
+                line,
+                line.replace(text_replacement, '').rstrip() + '\n')
+        else:
+            diff.delete_line(line_number)
```

Comment:
hm, `if in_multi_line_comment and not line_before_todo_comment` this branch will be run and might destroy the comment, right?